src/ConformalTracking.cc

```diff
@@ -621,12 +627,28 @@ void ConformalTracking::processEvent(LCEvent* evt) {
 
   // Finally reconstruct displaced tracks
 
-  Parameters displacedParameters(m_maxCellAngle * 10., m_maxCellAngleRZ * 10., m_chi2cut * 10., 5, 0.015, false, true);
-
   std::vector<int> allHits = {0, 1, 2, 3, 4, 5};
   stopwatch->Start(false);
   combineCollections(kdClusters, nearestNeighbours, allHits, collectionClusters);
-  buildNewTracks(conformalTracks, kdClusters, nearestNeighbours, displacedParameters, true, false);
+  double factor          = 10;
+  bool   caughtException = false;
+  do {
+    caughtException = false;
+    try {
+      Parameters displacedParameters(m_maxCellAngle * factor, m_maxCellAngleRZ * factor, m_chi2cut * factor, 5, 0.015, false,
+                                     true);
+      streamlog_out(DEBUG8) << "Building new tracks displaced: " << factor << std::endl;
+      buildNewTracks(conformalTracks, kdClusters, nearestNeighbours, displacedParameters, true, false);
+    } catch (TooManyTracksException& e) {
+      streamlog_out(MESSAGE) << "caught too many tracks, tightening parameters" << std::endl;
+      caughtException = true;
+      factor -= 1;
```

Comment:
how much is the reduction in number of tracks from factor = 10 to factor = 9? Isn't the -= 1 bin too fine?

Reply:
I will add a flag to disable the tightening of parameters, and if disabled directly skip the event.